api/src/test/java/io/kafbat/ui/service/rbac/MockedRbacUtils.java

```diff
@@ -0,0 +1,109 @@
+  public static final String DEV_CLUSTER = "dev";
+
+  public static final String TOPIC_NAME = "aTopic";
+  public static final String CONSUMER_GROUP_NAME = "aConsumerGroup";
+  public static final String SCHEMA_NAME = "aSchema";
+  public static final String CONNECT_NAME = "aConnect";
+
+  /**
+   * All actions to Resource.APPLICATIONCONFIG for dev and prod clusters.
+   *
+   * @return admin role
+   */
+  public static Role getAdminRole() {
+    Role role = new Role();
+    role.setName(ADMIN_ROLE);
+    role.setClusters(List.of(DEV_CLUSTER, PROD_CLUSTER));
+    Subject sub = new Subject();
+    sub.setType("group");
+    sub.setProvider("ldap");
+    sub.setValue("kafbat.group");
+    role.setSubjects(List.of(sub));
+    Permission applicationConfigPerm = new Permission();
+    applicationConfigPerm.setResource(Resource.APPLICATIONCONFIG.name());
+    applicationConfigPerm.setActions(List.of("all"));
+    List<Permission> permissions = List.of(
+        applicationConfigPerm
+    );
+    role.setPermissions(permissions);
+    role.validate();
+    return role;
+  }
+
+  /**
+   * View actions to topic, consumer, schema and connect.
+   *
+   * @return admin role
+   */
+  public static Role getDevRole() {
+    Role role = new Role();
+    role.setName(DEV_ROLE);
+    role.setClusters(List.of(DEV_CLUSTER));
+    Subject sub = new Subject();
+    sub.setType("group");
+    sub.setProvider("ldap");
+    sub.setValue("kafbat.group");
+    role.setSubjects(List.of(sub));
+    Permission topicViewPermission = new Permission();
+    topicViewPermission.setResource(Resource.TOPIC.name());
+    topicViewPermission.setActions(List.of(TopicAction.VIEW.name()));
+    topicViewPermission.setValue(TOPIC_NAME);
+
+    Permission consumerGroupPermission = new Permission();
+    consumerGroupPermission.setResource(Resource.CONSUMER.name());
+    consumerGroupPermission.setActions(List.of(ConsumerGroupAction.VIEW.name()));
+    consumerGroupPermission.setValue(CONSUMER_GROUP_NAME);
+
+    Permission schemaPermission = new Permission();
+    schemaPermission.setResource(Resource.SCHEMA.name());
+    schemaPermission.setActions(List.of(SchemaAction.VIEW.name()));
+    schemaPermission.setValue(SCHEMA_NAME);
+
+    Permission connectPermission = new Permission();
+    connectPermission.setResource(Resource.CONNECT.name());
+    connectPermission.setActions(List.of(ConnectAction.VIEW.name()));
+    connectPermission.setValue(CONNECT_NAME);
+
+    List<Permission> permissions = List.of(
+        topicViewPermission,
+        consumerGroupPermission,
+        schemaPermission,
+        connectPermission
+    );
+    role.setPermissions(permissions);
+    role.validate();
+    return role;
+  }
+
+  public static AccessContext getAccessContext(String cluster, Boolean resourceAccessible) {
+    AccessContext.ResourceAccess mockedResource = mock(AccessContext.ResourceAccess.class);
+    when(mockedResource.isAccessible(any())).thenReturn(resourceAccessible);
```

Comment:
`isAccessible` method implementation in `AccessContext` contains some crucial logic and testing this method inclusively makes a lot of sense for this kind of tests, so mocking this method kinda levels out the purpose of these tests, don't you think?

Reply:
Ah, nice! I forgot about `AccessContextTest` existence.